### ISSUE TITLE
Switch-Control design dont matches with Companion / has worng layout boundaries

### ISSUE THREAD

**Issue by jonas-klein** [mit-cml/appinventor-sources#2361]
**Describe the bug**

When i place in the designer a switch-Control with a long text, the text don't takes a line break. If the "Width"-property of the control ist set to "Fill parent...", the description text breaks, but the size of the layout boundaries of the control remains to the old one. In the companion-app, the rendering is flawless. Refer to the screenshots at the bottom.

**Affects**
- [X] Designer
- [ ] Blocks editor
- [ ] Companion
- [ ] Compiled apps
- [ ] Buildserver
- [ ] Debugging
- [ ] Other... (please describe)

**Expected behavior**
Flawless rendering in designer, too.

**Steps to reproduce**
Place a switch-control on the white area and add some long text to it. It's suffices, if there's no other controls placed. Appeared in Firefox 84.0.1 (64 bit).

> ![ai2 bug](https://user-images.githubusercontent.com/56542272/103469747-ff4b6200-4d68-11eb-87fe-c7d4a19424dc.png)

**Comment by ewpatton** [comment]
I believe when @SusanRatiLane implemented this we knew that this was an issue at the time and it has to do with how GWT calculates (or rather, doesn't) the height of the label. This is something that we might be able to address during the UI work.

**Comment by SusanRatiLane** [comment]
Yes, I remember this behavior, but I don't remember why it was challenging to fix. I'll get it in my queue to take a look.

**Comment by ewpatton** [comment]
IIRC, the issue was that the size of the label is computed outside of its DOM hierarchy, which allows it to be "infinitely" wide, so its height is always 1 line of text. This is then used by our layout calculations to position the items following the label. However, when placed in context the label does wrap because its width is no longer "infinite", so it takes up more space than used in the layout calculation so the button overlaps.

**Comment by patryk84a** [comment]
I think that the view in the designer is not that important and will never be 100% identical to the view on the phone. A trivial thing.
